pallets/attesters/src/lib.rs

```diff
@@ -0,0 +1,864 @@
+                        }
+                    }
+
+                    // Add the attestation to the last batch
+                    if let Some(last_batch) = target_batches.last_mut() {
+                        last_batch.attestations.push(sfx_attestation);
+
+                        if last_batch.attestations.len() >= T::MaxBatchSize::get() as usize {
+                            last_batch.status = BatchStatus::ReadyForSubmission;
+                        }
+                    }
+
+                    // Save the updated batches
+                    Batches::<T>::insert(target, target_batches);
+                },
+            }
+
+            Self::deposit_event(Event::AttestationSubmitted(account_id));
+
+            Ok(())
+        }
+
+        #[pallet::weight(10_000)]
+        pub fn commit_batch(
+            origin: OriginFor<T>,
+            target: [u8; 4],
+            batch_index: u32,
+        ) -> DispatchResult {
+            let submitter = ensure_signed(origin)?;
+
+            Batches::<T>::try_mutate(target, |batches_option| {
+                if let Some(batches) = batches_option {
+                    let index = batch_index as usize;
+                    if let Some(batch) = batches.get_mut(index) {
+                        match batch.status {
+                            BatchStatus::Pending | BatchStatus::ReadyForSubmission => {
+                                batch.status = BatchStatus::Committed;
+                                Ok(Some(batches.clone()))
+                            },
+                            _ => Err(Error::<T>::BatchAlreadyCommitted),
+                        }
+                    } else {
+                        Err(Error::<T>::BatchNotFound)
+                    }
+                } else {
+                    Err(Error::<T>::BatchNotFound)
+                }
+            })?;
+
+            // Reward the submitter
+            let fast_confirmation_cost =
+                FastConfirmationCost::<T>::get(target).unwrap_or_else(Zero::zero);
+            let total_reward = fast_confirmation_cost.saturating_mul(T::RewardMultiplier::get());
+
+            T::Currency::transfer(
+                &T::CommitmentRewardSource::get(),
+                &submitter,
+                total_reward,
+                ExistenceRequirement::KeepAlive,
+            )?;
+
+            Ok(())
+        }
+
+        #[pallet::weight(10_000)]
+        pub fn nominate(
+            origin: OriginFor<T>,
+            attester: T::AccountId,
+            amount: BalanceOf<T>,
+        ) -> DispatchResult {
+            let nominator = ensure_signed(origin)?;
+            Self::do_nominate(nominator.clone(), attester.clone(), amount)?;
+            Self::deposit_event(Event::Nominated(nominator, attester, amount));
+            Ok(())
+        }
+    }
+
+    impl<T: Config> Pallet<T> {
+        pub fn committee_size() -> usize {
+            T::CommitteeSize::get() as usize
+        }
+
+        pub fn do_nominate(
```

Comment:
Thoughts on pallet-staking from moonbeam? I looked through it, and to me it seems like it gives us what we need. Feel like there are a lot of things to mess up here.